solr/core/src/java/org/apache/solr/util/stats/InstrumentedHttpListenerFactory.java

```diff
@@ -86,38 +87,44 @@ private static String methodNameString(Request request) {
   }
 
   @Override
-  public RequestResponseListener get() {
-    return new RequestResponseListener() {
-      Timer.Context timerContext;
-      Span span = Span.getInvalid();
-
-      @Override
-      public void onQueued(Request request) {
-        // do tracing onQueued because it's called from Solr's thread
-        span = Span.current();
-        TraceUtils.injectTraceContext(request);
-      }
+  public final void onQueued(Request req) {
+    var listener = new PerRequestListener(req);
```

Comment:
this scenario is maybe a bit more awkward because on onQueued, we need to register a listener specific for this request because the listener has per-request state (fields).  This was previously happening for all listeners in Http2SolrClient but you'll see it gets removed there now.